### PR TITLE
Improving SDK types

### LINKED ISSUE #19285: [SDK, TypeScript] Filter type does not accept logical operators
### Describe the Bug

The [Filter type](https://docs.directus.io/reference/filter-rules.html) accepts only attributes of the entity it is filtering on, while it should also allow for [logical operators](https://docs.directus.io/reference/filter-rules.html#logical-operators).

This was reported first in an old issue https://github.com/directus/directus/issues/7475#issuecomment-1655762961.

### To Reproduce

```ts
 directus.request(
        readItems('users', {
          filter: {
            _or: [ //<- Type '{ _or: { name: string; }[]; }' is not assignable
              {
                name: 'martijn',
              },
              {
                name: 'directus',
              },
            ],
          },
        }),
      )
```

### Directus SDK Version

v11.0.0

### Hosting Strategy

Self-Hosted (Docker Image)

### LINKED ISSUE #19292: Cannot find module '@directus/schema' or its corresponding type declarations - Angular 16 - Ionic 7
### Describe the Bug

as Reported earlier in #19260 
This issue is not resolved. - Checked on @directus/sdk 11.0.1.
I am working on Angular project frontend only - Should I need to install knex, vue in it ?
can we decouple frontend type defs - Core SDK type defs?
am i doing something wrong ? Please suggest appropriate NPM package to use.
Let me know how can i contribute in resolving this.

256956839-565e0b00-d1f3-4e69-ab9b-1fb9ba473398

### To Reproduce

Fresh ionic app creation :
Ionic start --type=angular tabs

Install Directus sdk :
npm install @directus/sdk

Start the app - after importing sdk in appropriate service / component
ionic start

### Directus Version

10.5.1

### Hosting Strategy

Self-Hosted (Custom)

### LINKED ISSUE #19301: SDK GraphQL header text/plain
### Describe the Bug

Hello, i think i encounter a bug using the new SDK `@directus/sdk": "11.0.1"`.
When i run a query with GraphQL, the header `content-type` is `text/plain;charset=UTF-8` instead of `application/json;charset=UTF-8`

### To Reproduce

```js
async ql() {
    try {
        let directus = createDirectus(this.url).with(graphql());
        let q =
            `query   {
                info  {
                    welcome_message
                }
            }`
        return await directus.query(q);
    } catch (e) {
        console.log(e)
    }
}
```

The error i get is `"Invalid payload. Must provide query string."`

Rerunning the request with curl and changing the `content-type`, works as expected

### Directus Version

v10.4.3

### Hosting Strategy

Self-Hosted (Docker Image)

### PR DESCRIPTION
**Pull request by br41nslug**
Improving existing types and adding missing types.

- Added missing filter operators `_some/_none`/`_or/_and` (fixes #19285)
- removed the reliance on `@directus/types` (fixes #19292 and #19260)
- fixed content-type regression on graphql (fixes #19301)

### CODE REVIEW

**Review by rijkvanzanten — APPROVED** [review]

**Comment by rijkvanzanten** [comment]
Much simpler 👍🏻

File: .changeset/polite-buckets-raise.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+"@directus/sdk": patch
+---
+
+Added missing filter operators
+Removed the reliance on @directus/types
+Fixed content-type regression on graphql
```
Comment (paescuj) [inline .changeset/polite-buckets-raise.md:7]:
```suggestion
Added missing filter operators, removed the reliance on `@directus/types`, fixed content-type regression on graphql
```

**Review by paescuj — APPROVED** [review]
Looks great!

**Comment by br41nslug** [comment]
> > * removed the reliance on `@directus/types` (fixes [Cannot find module '@directus/schema' or its corresponding type declarations - Angular 16 - Ionic 7 #19292](https://github.com/directus/directus/issues/19292) and [Cannot find module '@directus/types' or its corresponding type declarations - Using angular 16 - Ionic 7 as Frontend #19260](https://github.com/directus/directus/issues/19260))
> 
> Should already be fixed with #19296, do we still want to remove reliance?

I did see that PR but am not 100% convinced that adding more dependencies is the right solution.  Considering there was only 1 type shared this seems like the most robust solution so i can keep focussing on the more important issues to fix.

**Comment by paescuj** [comment]
> I did see that PR but am not 100% convinced that adding more dependencies is the right solution. Considering there was only 1 type shared this seems like the most robust solution so i can keep focussing on the more important issues to fix.

Sure, makes sense for now. Just in case we want to add `@directus/types` back in the near future...

**Pull request merged**